### DISCUSSION TITLE
High memory usage for ADODG3_Wing

### DISCUSSION THREAD

**Discussion by benoitpaillard** [mdolab/dafoam#682] [Bug] [answered]
Hi all,

When running ADODG3_Wing tutorial on 12 cores, memory usage goes beyond the 64G memory that is fitted in my server and which has always be way more than enough for any CFD project.

FAQ mentions 100G of memory for 1e6 meshes, can you elaborate ? Is that a strong constraint ? Usually OpenFOAM cases take at most 2G per million cells.

**Comment by friedenhe** [comment, marked as answer]
What you observed is normal. Our _**discrete**_ adjoint solution needs a lot of memory than the CFD solution. We do not find this as a significant limitation because many modern HPC systems have enough memory to run large cases.
